src/DataflowPass.cpp

```diff
@@ -168,79 +205,132 @@ void DataflowPass::transfer(Instruction *instruction,
       bool loopBroken = false;
       for (auto Pair : this->memoryFunctions) {
         if (fnName == Pair.first) {
-          logout("not calling onalloc for "
-                 << arg << " of fnName " << fnName << " and inst "
-                 << *instruction << " and just returning now") return;
+          logout("not calling onalloc for " << arg << " of fnName " << fnName
+                                            << " and inst " << *instruction
+                                            << " and just returning now");
+          return;
 
         } else if (fnName == Pair.second) {
-          MethodsSet *methods =
-              inputProgramPoint.getPVRef(arg, false)->getMethodsSetRef();
-          logout("GETMAIN-5 ON '"
-                 << arg
-                 << "'") this->onDeallocationFunctionCall(methods, fnName);
+          MethodsSet *methods = pv->getMethodsSetRef();
+          logout("GETMAIN-5 ON '" << arg << "'");
+          this->onDeallocationFunctionCall(methods, fnName);
           loopBroken = true;
           break;
         }
       }
 
-      if (loopBroken) {
+      if (loopBroken || fnName == "llvm.dbg.declare") {
         continue;
       }
 
-      if (fnName == "llvm.dbg.declare") {
-        continue;
+      // finds annotations
+      logout("finding annotations for " << fnName);
+
+      /*
+        LLVM uses sret attributes as an optimization if the return type is
+        large. a struct with more than 2 fields can trigger this optimization.
```

Comment:
I think "more than 2" here can also mean "any array or struct field"

Reply:
which technically might just be one (the array/struct), but it has more than a single element